table/block_based_table_reader.cc

```diff
@@ -1119,10 +1119,16 @@ bool BlockBasedTable::PrefixMayMatch(const Slice& internal_key) {
     return true;
   }
 
+  ParsedInternalKey parsed_internal_key;
+  if (!ParseInternalKey(internal_key, &parsed_internal_key)) {
+    // This shouldn't happen, but if it does just stumble forward.
+    return true;
+  }
+
   assert(rep_->ioptions.prefix_extractor != nullptr);
   auto prefix = rep_->ioptions.prefix_extractor->Transform(
-      ExtractUserKey(internal_key));
-  InternalKey internal_key_prefix(prefix, 0, kTypeValue);
+      parsed_internal_key.user_key);
+  InternalKey internal_key_prefix(prefix, parsed_internal_key.sequence, kTypeValue);
```

Comment:
Let's also wait for @siying 's feedback, he has more experience around our table code.